src/components/LinkAdapter/LinkAdapter.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react"
+import { useTheme } from "@emotion/react"
+import styled from "@emotion/styled"
+
+const PlainLink = styled.a({
+  color: "inherit",
+  textDecoration: "none",
+})
+
+/**
+ * LinkAdapterPropsOverrides can be used with module augmentation to provide
+ * extra props to ButtonLink.
+ *
+ * For example, in a NextJS App, you might set `next/link` as your default
+ * Link implementation, and use LinkAdapterPropsOverrides to provide
+ * `next/link`-specific props.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+interface LinkAdapterPropsOverrides {}
```

Comment:
Really nice approach and a lot better than supporting a potentially growing list of peer dependencies.

It would be good to add an example of module augmentation (it took me a moment to figure it out). I think this works for React Router links that have a `to` prop, for example: 

```
declare module "@mitodl/smoot-design" {
  interface LinkAdapterPropsOverrides {
    to: string
  }
}
```